dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java

```diff
@@ -1200,4 +1201,41 @@ public void testAppToolCollections() throws Exception {
         Collection collection = organizationsApiAdmin.getCollectionById(registeredOrganization.getId(), createdCollection.getId());
         assertTrue((collection.getEntries().stream().anyMatch(entry -> Objects.equals(entry.getId(), appTool.getId()))));
     }
+
+    @Test
+    public void testDifferentLanguagesWithSameWorkflowName() throws Exception {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false);
+        final io.dockstore.openapi.client.ApiClient webClient = getOpenAPIWebClient(BasicIT.USER_2_USERNAME, testingPostgres);
+        io.dockstore.openapi.client.api.WorkflowsApi workflowClient = new io.dockstore.openapi.client.api.WorkflowsApi(webClient);
+        io.dockstore.openapi.client.api.UsersApi usersApi = new io.dockstore.openapi.client.api.UsersApi(webClient);
+
+        try {
+            workflowClient.handleGitHubRelease("refs/heads/differentLanguagesWithSameWorkflowName", installationId, workflowDockstoreYmlRepo, BasicIT.USER_2_USERNAME);
+            Assert.fail("should have thrown");
+        } catch (io.dockstore.openapi.client.ApiException ex) {
+            String message = ex.getMessage().toLowerCase();
+            assertTrue(message.contains("descriptor language"));
+            assertTrue(message.contains("workflow"));
+            assertTrue(message.contains("version"));
+        }
+        
+        // There should be one failure message
+        List<io.dockstore.openapi.client.model.LambdaEvent> events = usersApi.getUserGitHubEvents("0", 10);
+        assertEquals(0, events.stream().filter(lambdaEvent -> lambdaEvent.isSuccess()).count());
+        assertEquals(1, events.stream().filter(lambdaEvent -> !lambdaEvent.isSuccess()).count());
+        io.dockstore.openapi.client.model.LambdaEvent event = events.stream().filter(lambdaEvent -> !lambdaEvent.isSuccess()).findFirst().get();
+        String message = event.getMessage().toLowerCase();
+        assertTrue(message.contains("descriptor language"));
+        assertTrue(message.contains("workflow"));
+        assertTrue(message.contains("version"));
+
+        // No workflow should have been created.
+        // This will change in 1.13, wherein .dockstore.yml processing changes so that an error in one entry will not roll back the entire update, and a workflow with one version should have been created.
+
+        try {
+            io.dockstore.openapi.client.model.Workflow workflow = workflowClient.getWorkflowByPath("github.com/" + workflowDockstoreYmlRepo, WorkflowSubClass.BIOWORKFLOW, "versions,validations");
```

Comment:
Should `Assert.fail` if the workflow is created successfully

Reply:
good catch